docs/projects/tad/adrs/0005-ai-verify-technical-tests.md

```diff
@@ -0,0 +1,46 @@
+# TAD-0005 Use AI Verify for technical tests
+
+## Context
+
+The [AI Verify project](https://github.com/aiverify-foundation/aiverify) is set up in a modular way, and the
+technical tests is one of the modules. The AI Verify
+team is developing a feature which makes it possible to run the technical tests using an API, by using
+a Python file calling a method to run a test and providing the required configuration; for example, which
+model and dataset to use and some test specific configuration. The result of the test are returned in a JSON format,
+which can be processed in any way we please, like writing it to a file or System Card or store it in a database.
+
+### Pros
+
+* We have several technical tests we can use of the shelf, like SHAP, ALE, or Fairness metrics.
+* Tests are set up in a generic way using interfaces which allows others, like ourselves, to create their own plugins.
+* Loading models, pipelines and data is done through the AI Verify toolkit, which does validation
+ and support checks. All of these should adhere to a specific format, which one should be aware of.
+
+### Cons
+
+* The outcome of an AI Verify test depends on the implementation chosen by the tool's developers. This means
+ there's less control over what data goes into the process report compared to the final result report.
+* Adding our own plugins may require adding AI Verify frontend blocks we don't use ourselves.
+* We are dependent on the AI Verify ecosystem for supported models and data formats.
+
+## Assumptions
+
+* We can wrap the API and other AI Verify requirements in a Docker image.
+* We can run the Docker image independently where we only have to provide the model, datasets and other
+ required configuration to run a test.
+* We can monitor the progress of a test in our TAD tool.
+* We can process the results of a test in our TAD tool.
+
+## Decision
+
+We will use the technical tests of AI Verify for the TAD tool.
```

Comment:
For me the loss in flexibility is not balanced out by the advantages. But if you think this is the way. its fine by me.

Reply:
I would add the added nuance that we try to use AI Verify as much as possible for Technical tests as this forces us to standardize the models and the datasets in their format and make use of an international library, but of course not always it will be possible and then we will build our own workarounds for it

Reply:
I changed the decision to: Our technical tests will include, but may extend beyond, those offered by AI Verify.

@berrydenhartog It is (now) stated we won't only use AI-Verify, which implies and means we *can* add other tests or solutions. AI Verify offers a strong base, with model loading (and validation), data loading (and validation), and running (several) technical tests "as is", almost "plug and play", with a clear Interface and modular approach which allows for adding "our own". Again, we are not limited, but hopefully "launched" by using what is offered.

Reply:
Good to add this nuance, it is exactly what I had in mind, but it wasn't explicit yet.